### ISSUE TITLE
Extend --delete-files to allow file input in addition to globbing

### ISSUE THREAD

**Issue by jhabdas** [rtyley/bfg-repo-cleaner#218]
The `--delete-files` command accepts a `<glob>`. The `--strip-blobs-with-ids` command accepts a `<blob-ids-file>`. Proposal here is to enhance `--delete-files` to take a mixed argument value whereby a file containing a list of filenames, such as those produced ETL logging operations, so a specific set of files may be removed by the BFG.

Achieving this may resolve #51, as I believe the requestor may be trying to accomplish the same thing but doing it in a roundabout way.

**Comment by javabrett** [comment]
BFG can currently only remove files and folders by their "local" name, not their full name/path, a limitation discussed in #12.  So for this feature to be implemented, either that capability would need to be added natively (best), or some read-and-convert-to-object-ids capability be added, which would replace the home-grown external scripting currently sometimes done to achieve the same full-and-distinct-path deletion capability.

**Comment by jhabdas** [comment]
Skimmed #12. Unlike Jason my use case does not care about directories, which I believe should alleviate any performance issues.

> So for this feature to be implemented, either that capability would need to be added natively (best)

I'd settle for a homegow, but I thought we might be able to create something more potent, if that's best.

Leaving files behind for me is a nice to have. But I cannot lob off an entire folder at this point. So something in between a scalpel and machete should suffice.

**Comment by javabrett** [comment]
I was possibly not very clear with my previous comment.

BFG `--delete-files` can currently delete files by filename or glob, e.g. delete `1.txt` or `*.tgz`, but you currently cannot tell it to do-so within specific paths e.g. delete `/a/b/1.txt` but retain `/c/d/1.txt`, or only delete `/some/path/*.tgz`.  Likewise `--delete-folders` can only work with a folder-name, not a specific path.

Both of these have to do with the way BFG currently works recursively internally, and with the way it calls JGit.  It is currently focused on cleaning trees, and doesn't retain context of the entire path to that sub-tree, nor look it up using JGit.  So the routing performing the deletion only currently sees the local names and not the paths.

Several BFG users have used external scripting to work-around that, converting a input list of fully-qualified paths to blob object IDs, so they can be deleted with precision using `--strip-blobs-with-ids`, which works well.  What I was trying to point-out is that if you want to truly improve the way that works with BFG, you would be looking to add native support for full-path processing.  Otherwise you would just be moving that scripting from external to within BFG.

Who knows, maybe that will be the simplest way of doing it - converting paths to object IDs.  But when I looked at it, it seemed feasible to add path-context to the current code and be able to process full-paths as strings without resorting to pre-building a blob-list.

**Comment by jhabdas** [comment]
Oh no what you said was perfectly clear to me.

If `--delete-files` can only accept one filename when globbing is not used though, this still feels like a potential enhancement WRT #51 and/or your enhancements linked within.

I'd be satisfied just calling BFG multiple times in a loop. I'm just not certain that'd achieve the expected result (though it may). Thoughts?

**Comment by javabrett** [comment]
Ah I think I see now, you might be after #137.
